src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/IdamRoleMappingServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.cwrdapi.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.cwrdapi.domain.CaseWorkerIdamRoleAssociation;
+import uk.gov.hmcts.reform.cwrdapi.repository.CaseWorkerIdamRoleAssociationRepository;
+import uk.gov.hmcts.reform.cwrdapi.service.IdamRoleMappingService;
+
+import java.util.List;
+import java.util.Set;
+import javax.transaction.Transactional;
+
+@Service
+public class IdamRoleMappingServiceImpl implements IdamRoleMappingService {
+    @Autowired
+    CaseWorkerIdamRoleAssociationRepository cwIdamRoleAssocRepository;
+
+    /**
+     * Builds the idam role mappings for case worker roles.
+     * @param caseWorkerIdamRoleAssociations list of CaseWorkerIdamRoleAssociation
+     */
+    @Override
+    public void buildIdamRoleAssociation(
+            List<CaseWorkerIdamRoleAssociation> caseWorkerIdamRoleAssociations) {
+        cwIdamRoleAssocRepository.saveAll(caseWorkerIdamRoleAssociations);
+    }
+
+    /**
+     * Deletes all the records for the service code provided.
+     * @param serviceCode set of service code
+     */
```

Comment:
I think @Transactional should be on service in your case delete and save all should be part of same transaction right?

Reply:
I did get an issue for delete alone, so I have put in only on delete. I can move it to service. no issues 👍

Reply:
The reason why I have put only on delete is that, if suppose delete is successful and save all is failure then I don't want to rollback the delete ones as it should be deleted whenever we get new list. what's your thought on this?

Reply:
I think both should be one transaction.